CONTRIBUTING.md

```diff
@@ -224,9 +224,19 @@ The design review process is as follows:
   if (GVFSPlatform.Instance.KernelDriver.EnumerationExpandsDirectories)
   ```
 
-- *Add new interfaces when it makes sense for the product, not simply for unit testing*
+- *Testability of a class should be a priority in the design of a class and during code reviews.*
 
-  When a class needs to be mocked (or have a subset of its behavior mocked), prefer using virtual methods instead of adding a new interface.  VFS4G uses interfaces when multiple implementations of the interface exist in the product code.
+  When writing new code, aim to follow existing coding guidelines. However, when exiting guidelines and testability goals of a design conflict, optimize for testability.
+
+- *When writing tests, prefer a well defined boundary for what dependencies are test classes and what dependencies are product classes.*
+
+  When a test needs to control a dependency, try to implement test class from a contract instead of extending a product implementation class. This will ensure that we understand what behaviors are test classes and what are product classes.
+
```

Comment:
Is there anything we can add to encourage developers to add both "true unit tests" and "integration" unit tests?

I think it makes sense to always recommend that the true unit tests be added, and with possible/sensible also add unit tests that exercise method calls across the components of the product.